batcher/aligned-batcher/src/lib.rs

```diff
@@ -1189,39 +1199,67 @@ impl Batcher {
         proof_submitters: Vec<Address>,
         fee_params: CreateNewTaskFeeParams,
     ) -> Result<TransactionReceipt, BatcherError> {
-        info!("Creating task for: 0x{}", hex::encode(batch_merkle_root));
-
-        match try_create_new_task(
-            batch_merkle_root,
-            batch_data_pointer.clone(),
-            proof_submitters.clone(),
-            fee_params.clone(),
-            &self.payment_service,
-        )
-        .await
-        {
-            Ok(receipt) => Ok(receipt),
-            Err(BatcherSendError::TransactionReverted(err)) => {
-                // Since transaction was reverted, we don't want to retry with fallback.
-                warn!("Transaction reverted {:?}", err);
-
-                Err(BatcherError::TransactionSendError)
-            }
-            Err(_) => {
-                let receipt = try_create_new_task(
+        let result = retry_function(
+            || {
+                create_new_task_retryable(
                     batch_merkle_root,
-                    batch_data_pointer,
-                    proof_submitters,
-                    fee_params,
+                    batch_data_pointer.clone(),
+                    proof_submitters.clone(),
+                    fee_params.clone(),
+                    &self.payment_service,
                     &self.payment_service_fallback,
                 )
-                .await?;
-
-                Ok(receipt)
+            },
+            DEFAULT_MIN_RETRY_DELAY,
+            DEFAULT_BACKOFF_FACTOR,
+            DEFAULT_MAX_RETRIES,
+        )
+        .await;
+        match result {
+            Ok(receipt) => Ok(receipt),
+            Err(RetryError::Permanent(BatcherError::ReceiptNotFoundError)) => {
+                self.override_created_task(fee_params.gas_price).await;
+                Err(BatcherError::ReceiptNotFoundError)
             }
+            Err(_) => Err(BatcherError::TransactionSendError),
         }
     }
 
+    pub async fn override_created_task(&self, task_gas_price: U256) {
```

Comment:
small nit, I think it makes the intent a little clearer
```suggestion
    pub async fn cancel_create_new_task_tx(&self, task_gas_price: U256) {
```

Reply:
is the `task_gas_price` argument the bumped price to cancel the old one? or is it the old gas price? Let's make the argument name a little clearer about this

Reply:
it might make sense to ask for the current gas price again inside the function and bump it a little to make sure that the cancelling tx is included rather than using the old gas price

Reply:
We could also try indefinitely bumping the gas price until it gets accepted, at some point, it should be accepted, this way we would avoid flushing the whole queue. What do you think @entropidelic?

Reply:
@MarcosNicolau yes I agree, the cancel transaction should be included, if not the processing of tasks will get locked. A similar logic to what you did in the aggregator bump PR could be implemented 👍

Reply:
Made `cancel_create_new_tx` infinitely retryable!